lib/modules/datasource/datasource.spec.ts

```diff
@@ -18,7 +18,7 @@ class TestDatasource extends Datasource {
     } catch (err) {
       this.handleGenericErrors(err);
     }
-    return Promise.resolve(undefined);
+    return Promise.resolve(undefined as never);
```

Comment:
use `null`, is it's the allowed value
https://github.com/renovatebot/renovate/blob/9e28ef32752536501af3fbff8a63ccbe5343591c/lib/modules/datasource/types.ts#L78